velox/expression/SimpleFunctionAdapter.h

```diff
@@ -237,6 +242,11 @@ class SimpleFunctionAdapter : public VectorFunction {
       auto* oneUnpacked = packed.at(POSITION);
       auto oneReader = VectorReader<arg_at<POSITION>>(oneUnpacked);
 
+      if constexpr (FUNC::udf_has_callNullFree) {
+        oneReader.setChildrenMayHaveNulls();
```

Comment:
Actually now that I am looking at the code here, 
looks like we would always call   
applyContext.mayHaveNullsRecursive() and 
oneReader.setChildrenMayHaveNulls()
would it make sense to do the work of  setChildrenMayHaveNulls
while running mayHaveNullsRecursive().
some children wont be reachable due to early return in that case we dont care
as long as we check the cache MayHaveNulls in the same order.

Reply:
I had thought about that, the reason I didn't is that I thought it would be weird for a function like mayHaveNullsRecursive() to have side effects.  Especially since we rely on those side effects later in a performance critical way.

It would be easy (and currently correct) to think moving or removing the call to mayHaveNullsRecursive() would be safe (as long as applyContext.mayHaveNullsRecursive gets set somehow).  However, if it had that side effect it would suddenly become a critical function.

Reply:
make sense